sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumer.cs

```
@@ -182,6 +216,83 @@ protected EventHubConsumer()
+        ///
+        /// <returns>An <see cref="IAsyncEnumerable{T}"/> to be used for iterating over events in the partition.</returns>
+        ///
+        /// <remarks>
+        ///   Unlike calls to <see cref="ReceiveAsync"/>, each subscriber to events is presented with an independent iterator; if there are multiple
+        ///   subscribers, they will each receive their own copy of an event to process, rather than competing for them.
+        ///
+        ///   Subscriptions will still compete with <see cref="ReceiveAsync" /> calls, however.  It is recommended that consumers either subscribe to
+        ///   events or explicitly receive batches, but do not use both approaches concurrently.
+        /// </remarks>
+        ///
+        /// <seealso cref="SubscribeToEvents(CancellationToken)"/>
+        ///
+        public virtual IAsyncEnumerable<EventData> SubscribeToEvents(TimeSpan? maximumWaitTime,
+                                                                     CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return CreateEmptyAsyncEnumerable<EventData>();
+            }
+
+            try
+            {
+                var maximumQueuedEvents = Math.Min((Options.PrefetchCount / 4), (BackgroundPublishReceiveBatchSize * 2));
+                var subscription = SubscribeToChannel(EventHubPath, PartitionId, ConsumerGroup, maximumQueuedEvents, cancellationToken);
+
+                return new ChannelEnumerableSubscription<EventData>(subscription.ChannelReader, maximumWaitTime, () => UnsubscribeFromChannelAsync(subscription.Identifier), cancellationToken);
+            }
+            catch (Exception ex) when
+                (ex is TaskCanceledException
+                || ex is OperationCanceledException)
+            {
```

Why suppress the exception?

---

They're expected when cancellation is requested during semaphore acquisition or a `Task.Delay`; they aren't an actual exception case.   I don't believe that they should be surfaced; we should follow our normal clean up for when cancellation is triggered.